layouts/shortcodes/readfile.md

```diff
@@ -1,11 +1,119 @@
-{{ if eq (.Get "relative") "true" }}
+{{/*
+
+Overrides Hugo's readFile (https://gohugo.io/functions/readfile/).
+
+Use this `readfile` shortcode to include the contents of a file into another file.
+Specify the `readfile` shortcode in a specific location within the "parent"
+file to "include" content from the source file.
+
+Default behavior is to obtain source files in a directory that is relative to
+the parent file. See details below.
+
+IMPORTANT: You cannot nest shortcodes. When the Hugo build runs on a file,
+any shortcodes within that file are processed. For example, if you use `readFile`
+to include a source file, and that source file contains another shortcode,
+none of those "nested shortcodes" are processed. Nesting a shortcode within another
+shortcode results in incorrectly rendered content
+(ie. `{{% the short code syntax shows in the HTML of the parent file %}}`).
+
+Syntax:
+    * For Markdown: Use %, `{{%...%}}`. This sends the source file to the Markdown
+      processor and also enables the in-page TOC (in the right-side nav).
+
+      Examples
+        * Include a `README.md` in the same folder's `_index.md` file:
+          `{{% readfile file="index.md" %}}`
+
+          Important: For SEO, all README.md files that are not used by the required
+          `_index.md` (Hugo section definition) are renamed to index.md at build time.
+          If you want to use a `readFile` for some other `README.md` file
+          (a `README.md` file is not used by and `_index.md` section def),
+          you must specify `index.md` (no underscore).
+
+        * Include any Markdown file, like `shared-content.md`, into another
+          parent file that lives in the same folder:
+          `{{% readfile file="shared-content.md" %}}`
+
+    * For HTML: Use < >, {{<...>}}. This prevents Hugo from sending the content to
+      the Markdown processor and copies in the content as is.
+
+      * Example of an HTML source file in same folder as the parent file:
```

Comment:
Indent here is strange (different than "Examples" above).

Reply:
fixed